item-nbt-plugin/src/main/java/de/tr7zw/nbtapi/plugin/tests/compounds/IteratorTest.java

```diff
@@ -0,0 +1,65 @@
+package de.tr7zw.nbtapi.plugin.tests.compounds;
+
+import de.tr7zw.changeme.nbtapi.NBTContainer;
+import de.tr7zw.changeme.nbtapi.NBTList;
+import de.tr7zw.changeme.nbtapi.NbtApiException;
+import de.tr7zw.nbtapi.plugin.tests.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class IteratorTest implements Test {
+
+    @Override
+    public void test() throws Exception {
+        NBTList<Integer> testList;
+
+        testList = initIntegerList();
+        testIterator(testList.iterator());
+
+        testList = initIntegerList();
+        testIterator(testList.listIterator());
+    }
+
+    private NBTList<Integer> initIntegerList() {
+        NBTContainer comp = new NBTContainer();
+        NBTList<Integer> list = comp.getIntegerList("test");
+        list.add(1);
+        list.add(2);
+        list.add(3);
+        list.add(4);
+        return list;
+    }
+
+    private static void testIterator(Iterator<Integer> iterator) {
+        assertTrue(iterator.hasNext());
+        assertTrue(iterator.next() == 1);
+        assertTrue(iterator.hasNext());
+        assertTrue(iterator.next() == 2);
+        iterator.remove();
+        assertTrue(iterator.hasNext());
+        assertTrue(iterator.next() == 3);
+        assertTrue(iterator.hasNext());
+        assertTrue(iterator.next() == 4);
+        testNoMoreElements(iterator);
+    }
+
+    private static void testNoMoreElements(Iterator<Integer> iterator) {
+        assertTrue(!iterator.hasNext());
```

Comment:
May want to replace that with a normal check and a NbtApiException, I think asserts wheren't a good idea

Reply:
These `assertTrue` do actually throw `NbtApiException`